app/src/main/java/com/github/swent/echo/compose/association/AssociationDetails.kt

```diff
@@ -1,18 +1,169 @@
+                    stringResource(R.string.association_details_description),
+                    modifier = Modifier.align(Alignment.Center),
+                    style = MaterialTheme.typography.titleSmall
+                )
+            }
+            Box(
+                modifier =
+                    Modifier.weight(weightItems)
+                        .padding(paddingItems)
+                        .height(tabHeight)
+                        .clickable { associationDetailsTab = AssociationDetailsTab.EVENTS }
+                        .testTag("association_details_events_tab")
+            ) {
+                Text(
+                    stringResource(R.string.association_details_events),
+                    modifier = Modifier.align(Alignment.Center),
+                    style = MaterialTheme.typography.titleSmall
+                )
+            }
+        }
+        Box(modifier = Modifier.fillMaxWidth()) {
+            when (associationDetailsTab) {
+                AssociationDetailsTab.DESCRIPTION -> {
+                    Box(
+                        modifier =
+                            underlineShape
+                                .align(Alignment.CenterStart)
+                                .padding(start = underlinePadding)
+                                .background(MaterialTheme.colorScheme.primary)
+                                .testTag("association_details_underline_description")
+                    )
+                }
+                AssociationDetailsTab.EVENTS -> {
+                    Box(
+                        modifier =
+                            underlineShape
+                                .align(Alignment.CenterEnd)
+                                .padding(end = underlinePadding)
+                                .background(MaterialTheme.colorScheme.primary)
+                                .testTag("association_details_underline_events")
+                    )
+                }
+            }
+        }
+        when (associationDetailsTab) {
+            AssociationDetailsTab.DESCRIPTION -> {
+                Spacer(modifier = Modifier.height(verticalSpace))
+                // Text(association.largeDescription)
+                Text(
+                    association.description,
+                    modifier = Modifier.testTag("association_details_description_text")
+                )
+                /*
+                Spacer(modifier = Modifier.height(verticalSpace))
+                Text(
+                    "Contacts:",
+                    style = MaterialTheme.typography.titleSmall,
+                    modifier = Modifier.padding(5.dp)
+                )
+                Text(association.url, modifier = Modifier.padding(5.dp))
+                 */
+            }
+            AssociationDetailsTab.EVENTS -> {
+                Spacer(modifier = Modifier.height(verticalSpace))
+                ListDrawer(events, "", "", isOnline)
+            }
+        }
+    }
```

Comment:
The readability of this section could be improved. I believe there's two possible ways to do this:
1. Make a big switch on `associationDetailsTab`. This would have the benefit of being easy to read, but would lead to a lot of duplicated code, given the similarities between the two modes.
2. Given that the two modes share the same structure, make several small switches at the beginning to set the values that differ. For example,
```
val alignment = when (associationDetailsTab) {
    DESCRIPTION -> Alignment.CenterStart
    EVENTS -> Alignment.CenterEnd
}
```

Reply:
That's a very good point ! I struggled to find a solution to this problem, that's some great idea you gived to me ! I'll create an issue for the next sprint in order to apply those changes ;)

Reply:
Here is the future issue : https://github.com/orgs/echo-swent/projects/1/views/3?pane=issue&itemId=63254196